shared/modules/network.utils.ts

```diff
@@ -56,3 +57,20 @@ export function isTokenDetectionEnabledForNetwork(chainId: string | undefined) {
 function isSafeInteger(value: unknown): value is number {
   return Number.isSafeInteger(value);
 }
+
+export async function shouldShowLineaMainnet(rpcUrl: string): Promise<boolean> {
+  const currentDateHigherThanReleaseDate =
+    new Date().getTime() > Date.UTC(2023, 6, 11);
+
+  if (!currentDateHigherThanReleaseDate) {
+    return false;
+  }
+
+  let chainId: string | null;
+  try {
+    chainId = (await jsonRpcRequest(rpcUrl, 'eth_chainId')) as string;
```

Comment:
What is the purpose of this chain ID check?

Reply:
It just to check if the linea mainnet rpc url is working. We can choose another endpoint if you want :)

Reply:
good point

Reply:
I think there might be a product decision to make here though. Effectively the network would be hidden at launch for anyone not on an infura network. How many people do we think that is, and is that number acceptable given the goals of the network launch?

Reply:
There is an alternative, that would require product approval, where we should the network after the date and only make a network request once the user attempts to switch to the linea network, and if it fails we could show a custom error message (e.g. "Thanks for your interest in Linea Mainnet, it will be available soon!"), or hide the network from the list or just fallback to our current failure mode when failing to connect to a network (loading spinner and then the option to try again or switch networks)

Reply:
That seems like a good idea to me!